File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        sort(nums);
        return nums[nums.length/2];
    }
    
        static void sort(int[] arr){
        boolean swap;
        for (int i = 0; i<arr.length; i++){
            swap = false;
            for (int j = 1; j < arr.length-i; j++) {
                if(arr[j-1]>arr[j]){
                    int temp = arr[j];
                    arr[j] = arr[j-1];
                    arr[j-1] = temp;
                    swap = true;
                }
            }
            if(!swap){
                break;
            }
        }
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int el;
        for(int i = 0; i<nums.size(); i++){
            if(count == 0){
                count = 1;
                el = nums[i];
            }else if(nums[i] == el){
                count ++;
            }else{
                count --;
            }
        }
        
        int c = 0;
        for(int i = 0; i<nums.size(); i++){
            if(nums[i] == el) c++;
        }
        if(c>nums.size()/2) return el;
        return -1;
    }
};